### PR TITLE
Create UniswapNFTManager so users can import NFTs as collateral

### PR DESCRIPTION
**Pull request by haydenshively**

### CODE REVIEW

File: periphery/src/managers/UniswapNFTManager.sol [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        );
+
+        NFT_MANAGER.collect(
+            INFTManager.CollectParams({
+                tokenId: tokenId,
+                recipient: recipient,
+                amount0Max: type(uint128).max,
+                amount1Max: type(uint128).max
+            })
+        );
```
Comment (IanWoodard) [inline periphery/src/managers/UniswapNFTManager.sol:90]: I assume we opt not to burn it to increase liquidity later on. Does this sound right?

File: periphery/src/managers/UniswapNFTManager.sol [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            NFT_MANAGER.increaseLiquidity(
+                INFTManager.IncreaseLiquidityParams({
+                    tokenId: tokenId,
+                    amount0Desired: burned0,
+                    amount1Desired: burned1,
+                    amount0Min: 0,
+                    amount1Min: 0,
+                    deadline: block.timestamp
+                })
+            );
```
Comment (IanWoodard) [inline periphery/src/managers/UniswapNFTManager.sol:68]: The reason we don't burn the NFT, correct?

File: periphery/src/managers/UniswapNFTManager.sol [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        if (liquidity < 0) {
+            // safety checks since this contract will be approved to manager users' positions
+            {
+                require(FACTORY.isBorrower(msg.sender));
+                (address owner, , ) = borrower.slot0();
+                require(owner == NFT_MANAGER.ownerOf(tokenId));
+            }
+
+            _withdrawFromNFT(tokenId, uint128(-liquidity), msg.sender);
+            borrower.uniswapDeposit(lower, upper, uint128(-liquidity));
```
Comment (IanWoodard) [inline periphery/src/managers/UniswapNFTManager.sol:46]: Should this be negative?

**Review by IanWoodard — APPROVED** [review]

File: periphery/src/managers/UniswapNFTManager.sol (hunk 1 above)
Comment (haydenshively) [reply periphery/src/managers/UniswapNFTManager.sol:90]: Exactly!

File: periphery/src/managers/UniswapNFTManager.sol (hunk 3 above)
Comment (haydenshively) [reply periphery/src/managers/UniswapNFTManager.sol:46]: Yes. The sign of `liquidity` determines whether we move liquidity from NFT to Borrower, or Borrower to NFT. Negative means into Borrower, but once we've checked the sign all the actual deposit logic expects a `uint128`.

**Pull request merged**